packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -561,6 +625,20 @@ const extractTemplateLiteral = (node: t.TemplateLiteral, meta: Metadata): CSSOut
       return acc + quasi.value.raw + suffix;
     }
 
+    /**
+     * Manipulate the node if the CSS property, pseudo classes or pseudo elements are defined
+     * within a template element rather than in a expression.
+     * Eg :hover { color: ${({ isPrimary }) => (isPrimary ? 'green' : 'red')}; }
+     */
+    if (
+      t.isArrowFunctionExpression(nodeExpression) &&
+      t.isConditionalExpression(nodeExpression.body) &&
+      isCssPropertyInTemplateElement(quasi)
+    ) {
+      const nextQuasis = node.quasis[index + 1];
+      moveCssPropertyInExpression(nodeExpression.body, quasi, nextQuasis);
```

Comment:
Is it necessary to move the property in the expression? Can't you just directly add to `css`, `variables`, or `acc` when you find the property and value?

Reply:
@MonicaOlejniczak Currently we [expect the property to be defined within the expression](https://github.com/atlassian-labs/compiled/blob/master/packages/babel-plugin/src/utils/css-builders.tsx#L320).

Otherwise, given 
`color: ${({ isTrue }) => isTrue ? 'red' :  'green'};`
it will output 
```
cssOutput: {
css: [
{
css: 'red',
type: 'unconditional'
}
]}
```
and cause an error.

Are you suggesting to manipulate the css in [here](https://github.com/atlassian-labs/compiled/blob/master/packages/babel-plugin/src/utils/css-builders.tsx#L583), meaning that
- we let `extractConditionalExpression` evaluate all expressions, even without` :`
- add interpolation, if any, directly in the result so that we avoid creating an unconditional css for the blank CSS property?

For example, instead of having this output
![image](https://user-images.githubusercontent.com/14813563/153319787-7b30d991-8d7f-4d98-be88-89f31198bb5a.png)
we have
![image](https://user-images.githubusercontent.com/14813563/153320781-a8d77f47-3e80-4e4a-a96a-cb46f428d87d.png)

Reply:
Yeah so the alternative would be to build a new context in meta that is aware of the property from the template literal, handle the logic in `extractCondtionalExpression` and then adjust the output within the `extractTemplateLiteral` logic. I think it complicates nested selectors more, so don't worry about it for the time being.